content/blog/2021-02-01-tidynhl-analyse/index.fr.md

```diff
@@ -0,0 +1,710 @@
+- Je compare les attaquants avec les attaquants et les défenseurs avec les défenseurs.
+- En raison du plus petit nombre de gardiens repêchés, je ne considère pas de fenêtre pour cette position. Ainsi, pour qu'un gardien soit considéré comme un "bon choix", il doit avoir récolté plus de victoires que tous les gardiens repêchés après.
+- Je donne également le bénéfice du doute aux gardiens, j'applique encore une fois un ratio de **95%** sur le nombre de victoires.
+- Les gardiens doivent avoir récoltés au moins **50** victoires pour être considérés comme un "bon choix".
+
+Ces décisions un peu arbitraires sont basées sur mon jugement personnel. Je vous laisse le soin de changer certains de ces paramètres comme bon vous semble. Maintenant, voici la fonction qui nous permettra de tester notre approche.
+
+
+
+```r
+define_good_choice <- function(dt, player_window, player_min_pts, goalie_min_wins, interval) {
+  
+  dt[, good_pick := NA]
+
+  for (row in seq_len(nrow(dt))) {
+  
+    # On stock les informations sur le choix à valider
+    draft_year_temp <- dt[row,]$draft_year
+    draft_pick_temp <- dt[row,]$draft_overall
+    draft_position_temp <- dt[row,]$player_position_type
+    draft_nb_points_temp <- dt[row,]$skater_points
+    draft_wins_temp <- dt[row,]$goalie_wins
+  
+    if (draft_position_temp %in% c("F", "D")) {
+  
+      # On filtre les joueurs repêchés dans la fenêtre
+      dt_temp <- dt[draft_year == draft_year_temp & player_position_type == draft_position_temp & draft_overall <= (draft_pick_temp + player_window) & draft_overall > draft_pick_temp,]
+      
+      # On calcule le nombre de points maximal dans la fenetre
+      max_pts_windows <- max(dt_temp$skater_points, na.rm = TRUE)
+      
+      # Si manquant, on remplace par 0
+      max_pts_windows <- ifelse(is.na(max_pts_windows), 0, max_pts_windows)
+      draft_nb_points_temp <- ifelse(is.na(draft_nb_points_temp), 0, draft_nb_points_temp)
+  
+      if (draft_nb_points_temp > (max_pts_windows * interval) & draft_nb_points_temp >= player_min_pts) {
+        dt[row,]$good_pick <- TRUE
+      } else {
+        dt[row,]$good_pick <- FALSE
+      }
+  
+    } else if (draft_position_temp == "G") {
+  
+      # Aucune fenetre pour les gardiens
+      dt_temp <- dt[draft_year == draft_year_temp & player_position_type == draft_position_temp & draft_overall > draft_pick_temp,]
+      
+      # On calcule le nombre de victoires maximal après le choix
+      max_win_windows <- max(dt_temp$goalie_wins, na.rm = TRUE)
+      
+      # Si manquant, on remplace par 0
+      max_win_windows <- ifelse(is.na(max_win_windows), 0, max_win_windows)
+      draft_wins_temp <- ifelse(is.na(draft_wins_temp), 0, draft_wins_temp)
+  
+      if (draft_wins_temp > (max_win_windows * interval) & draft_wins_temp >= goalie_min_wins) {
+        dt[row,]$good_pick <- TRUE
+      } else {
+        dt[row,]$good_pick <- FALSE
+      }
+  
+    }
+  
+  }
+  
+  return(dt)
+  
+}
```

Comment:
Écoute, je suis sur que tu me vois venir avec mon tracteur, j'suis pas tant un fan des `for` loops, comme tout bon puriste R. Mais bon, j'irai avec cette quote biblique :

> Que celui qui n'a jamais péché lui jette la première pierre! (Jn 8,7)

Reply:
Première citation de la bible dans l'histoire des PR sur GitHub ...

Reply:
Mais blague a part, j'aime bien le `for` dans ce cas-ci perso, vu qu'il y a bcp d'opérations à chaque itération, je trouve ça explicite et facile à lire ...

Reply:
Ouais je comprends ton point. La méthode favorisée selon les geeks puristes serait quand même de looper avec un `mapply` dans lequel chaque colonne de ton `data.table` serait passé en argument. Mais comme je te dis, pas de stress avec ça hahaha